File: src/app/core/constants/fees.ts
/**
 * Fee rates shown in the order ticket — single source of truth.
 *
 * NOTE: this is the client-side order-preview fee model (a flat commission plus
 * VAT on the commission). It is deliberately separate from the backend
 * settlement fee schedule (buyer/seller/settlement basis points), which is
 * authoritative for actual charges. Kept here so the trading screen's estimate
 * isn't a set of bare magic numbers.
 */

export const FEES = {
  /** Commission as a fraction of trade value (0.10%). */
  COMMISSION_RATE: 0.001,
  /** VAT as a fraction of the commission (15%). */
  VAT_RATE: 0.15,
} as const;

File: backend/dotnet-services/src/GccBond.Shared/Constants/Fees.cs
namespace GccBond.Shared.Constants;

/// <summary>
/// Settlement fee schedule — single source of truth for the authoritative
/// charges applied when a trade settles. Expressed as fractions of notional
/// (basis points ÷ 10,000).
/// </summary>
public static class Fees
{
    /// <summary>Buyer commission — 25 bps (0.25%).</summary>
    public const decimal BuyerBps = 0.0025m;

    /// <summary>Seller commission — 25 bps (0.25%).</summary>
    public const decimal SellerBps = 0.0025m;

    /// <summary>Settlement fee — 10 bps (0.10%).</summary>
    public const decimal SettlementBps = 0.001m;
}
